tests/periph_i2c/tests/03__periph_i2c_error_codes.robot

```diff
@@ -0,0 +1,26 @@
+*** Settings ***
+Documentation       Data driven tests to verify writing to a valid register.
+
+Suite Setup         Run Keywords    Reset DUT and PHILIP
+...                                 API Firmware Should Match
+Test Setup          Run Keywords    Reset DUT and PHILIP
+...                                 API Firmware Should Match
+...                                 I2C Acquire
+Test Teardown       I2C Release
+
+Resource            periph_i2c.keywords.txt
+Resource            api_shell.keywords.txt
+
+Force Tags          periph  i2c
+
+*** Test Cases ***
+Wrong Address Should ENXIO
+    [Documentation]                         Verify address NACK occurs
+    API Call Should Error                   I2C Read Reg  addr=42
+    API Last Result Message Should Contain  ENXIO
+
+Data NACK Should Data EIO
+    [Documentation]                         Verify data NACK occurs
+    PHiLIP.write_and_execute                i2c.mode.nack_data  1
```

Comment:
you can omit underscores and use spaces instead, might look/read nicer and RF supports that by default

Reply:
Yes to invalid but we should distiguish between error codes. Our API states it for one, and there are practical reasons why one would need to know the difference between address NACK and data NACK.

Reply:
reply in the wrong place, should be for the last comment :disappointed:

Reply:
Done